src/components/app.jsx

```diff
@@ -50,10 +50,10 @@ const PublicContext = withClient(({ client }) => {
           setSharedDocumentId(id)
         } else {
           // public route = /public/?sharecode=xxxxx
-          // There is no id. It should be a sharecode
-          // dedicated to a unique note. We look into
-          // permissions and try to open the first file.
-          const { id, readOnly } = await getSharedDocument(client)
+          // There is an id since the sharecode is for the
+          // folder.
+          const { readOnly } = await getSharedDocument(client)
+          const id = searchParams.get('id')
```

Comment:
on aura quand même des docs sans paramètre id. Pour moi il faut gérer le fallback 
```suggestion
          const { defaultId, readOnly } = await getSharedDocument(client)
          const id = searchParams.get('id') || defaultId
```